peps/pep-0752.rst

```diff
@@ -0,0 +1,399 @@
+
+Root grants given to community projects SHALL always be public.
+
+.. _repository-metadata:
+
+Repository Metadata
+-------------------
+
+To allow installers and other tooling insight into this metadata for a given
+artifact upload of a namespaced package, the :pep:`JSON API <691>` MUST include
+the following keys:
+
+* ``namespace``: This is the associated `normalized <naming_>`_
+  namespace e.g. ``foo-bar``. If the namespace matches a child grant and the
+  user happens to be authorized for both the child and the root grant, this
+  MUST be the namespace associated with the child grant.
+* ``owner``: This is the organization with which the user is associated and
+  owner of the grant. If the namespace is `public <public-namespaces_>`_ and
+  the user is not part of the owner organization, this key MUST NOT be present.
+  This omission is useful for tools that wish to make a distinction between
+  official and community packages.
+
+The `Simple API`__ MAY include the aforementioned keys as attributes, for
+example:
+
+__ https://packaging.python.org/en/latest/specifications/simple-repository-api/#base-html-api
+
+.. code-block:: html
+
+    <a href="..." namespace="foo-bar" owner="org1">...</a>
+
+Grant Removal
+-------------
+
+If a grant is shared with other organizations, the owner organization MUST
+initiate a transfer as a prerequisite for organization deletion.
+
+If a grant is not shared, the owner may unclaim the namespace in either of the
+following circumstances:
+
+* The organization manually removes themselves as the owner.
+* The organization is deleted.
+
+When a reserved namespace becomes unclaimed, repositories:
+
+1. MUST remove the `visual indicator <user-interface_>`_
+2. MUST NOT modify past `release metadata <repository-metadata_>`_
+
+Grant Approval Criteria
+-----------------------
+
+Applications lacking substantial evidence of satisfying the following criteria
+are likely to be rejected:
+
+1. Does the namespace properly and clearly identify the reservation owner?
+2. Is the namespace something common that should not belong to any individual
+   owner or organization? Avoid namespaces that are shorter than four
+   characters and avoid common or generic words.
+3. Would *not* reserving the namespace cause ambiguity, confusion, or other
+   harm to the community?
+4. Is the organization actively using the namespace?
+
+By default, `root grant <root-grant_>`_ are only available for
+`corporate organizations <corp-orgs_>`_. Other organizations may be considered
+on a case-by-case basis, some examples:
+
+* Large, popular open-source projects with many packages like
+  `OpenTelemetry`__
+* Universities that actively publish packages
+* NPOs/NGOs that actively publish packages like
+  `Our World in Data <https://github.com/owid>`__
+* Government organizations that actively publish packages
+
+__ https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation
```

Comment:
This does not actually define approval criteria, but instead some "likely" reasons for rejection, and seems to implicitly define that namespaces are only available for corporations -- though again does not normatively define this.

Reply:
Also, should there be a requirement that a granted namespace match in any way the org name or can they be completely different?  (I guess the answer is yes they have to be different, since a company may own several orgs and namespace?).

Reply:
I think being the same as the organiation would soon be unenforceable (suppose an entity is legally registered as "L'Entreprise commerciale SARL" but traded as "Business Corp." in anglophone states, or e.g. Alphabet or Meta Platforms incs, which would perhaps like usage of the "google" or "instagram" names!). Likewise, the Sphinx project would probably want to register `sphinxcontrib-*` (if we were allowed!!)

Reply:
I revamped this entire section to use `MUST` and `SHOULD` for different criteria. The clear naming concern you mention is `SHOULD`.